lib/core/base/context.js

```diff
@@ -11,208 +12,6 @@ import {
-  if (
-    (context && typeof context === 'object') ||
-    context instanceof window.NodeList
-  ) {
-    if (context instanceof window.Node) {
-      return {
-        include: [context],
-        exclude: []
-      };
-    }
-
-    if (
-      context.hasOwnProperty('include') ||
-      context.hasOwnProperty('exclude')
-    ) {
-      return {
-        include:
-          context.include && +context.include.length
-            ? context.include
-            : [document],
-        exclude: context.exclude || []
-      };
-    }
-
-    if (context.length === +context.length) {
-      return {
-        include: context,
-        exclude: []
-      };
-    }
-  }
-
-  if (typeof context === 'string') {
-    return {
-      include: [context],
-      exclude: []
-    };
-  }
-
-  return {
-    include: [document],
-    exclude: []
-  };
-}
-
-/**
- * Finds frames in context, converts selectors to Element references and pushes unique frames
- * @private
- * @param  {Context} context The instance of Context to operate on
- * @param  {String} type     The "type" of thing to parse, "include" or "exclude"
- * @return {Array}           Parsed array of matching elements
- */
-function parseSelectorArray(context, type) {
-  var item,
-    result = [],
-    nodeList;
-  for (var i = 0, l = context[type].length; i < l; i++) {
-    item = context[type][i];
-    // selector
-    if (typeof item === 'string') {
-      nodeList = Array.from(document.querySelectorAll(item));
-      //eslint no-loop-func:0
-      result = result.concat(
-        nodeList.map(node => {
-          return getNodeFromTree(node);
-        })
-      );
-      break;
-    } else if (item && item.length && !(item instanceof window.Node)) {
-      if (item.length > 1) {
-        pushUniqueFrameSelector(context, type, item);
-      } else {
-        nodeList = Array.from(document.querySelectorAll(item[0]));
-        //eslint no-loop-func:0
-        result = result.concat(
-          nodeList.map(node => {
-            return getNodeFromTree(node);
-          })
-        );
-      }
-    } else if (item instanceof window.Node) {
-      if (item.documentElement instanceof window.Node) {
-        result.push(context.flatTree[0]);
-      } else {
-        result.push(getNodeFromTree(item));
-      }
-    }
-  }
-
-  // filter nulls
-  return result.filter(r => r);
-}
-
-/**
- * Check that the context, as well as each frame includes at least 1 element
- * @private
- * @param  {context} context
- * @return {Error}
- */
-function validateContext(context) {
-  if (context.include.length === 0) {
-    if (context.frames.length === 0) {
-      var env = respondable.isInFrame() ? 'frame' : 'page';
-      return new Error('No elements found for include in ' + env + ' Context');
-    }
-    context.frames.forEach((frame, i) => {
-      if (frame.include.length === 0) {
-        return new Error(
-          'No elements found for include in Context of frame ' + i
-        );
-      }
-    });
```

Comment:
I left this out because it never worked.We're returning in a forEach. That does nothing. This is completely pointless too. If include is empty, axe tests the entire frame.